backends-clickhouse/src/main/java/io/glutenproject/execution/RowToCHNativeColumnarExec.scala

```diff
@@ -0,0 +1,111 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.glutenproject.execution
+
+import io.glutenproject.vectorized._
+import org.apache.spark.broadcast
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.catalyst.expressions.{Attribute, SortOrder, UnsafeProjection}
+import org.apache.spark.sql.catalyst.plans.physical.Partitioning
+import org.apache.spark.sql.execution.SparkPlan
+import org.apache.spark.sql.vectorized.ColumnarBatch
+
+import java.util.concurrent.TimeUnit._
+
+/**
+ * this class is only for test use, not yet suggested to use in non-test code
+ * the performance is not optimized
+ * @param child
+ */
+case class RowToCHNativeColumnarExec(child: SparkPlan) extends RowToArrowColumnarExec(child = child) {
+
+  override def doExecuteColumnarInternal(): RDD[ColumnarBatch] = {
+    val numInputRows = longMetric("numInputRows")
+    val numOutputBatches = longMetric("numOutputBatches")
+    val processTime = longMetric("processTime")
+    // This avoids calling `schema` in the RDD closure, so that we don't need to include the entire
+    // plan (this) in the closure.
+    val localSchema = this.schema
+    child.execute().mapPartitions { rowIterator =>
+      val projection = UnsafeProjection.create(localSchema)
+      val cvt = new BlockNativeConverter
+      if (rowIterator.hasNext) {
+        val res = new Iterator[ColumnarBatch] {
+          private val byteArrayIterator = rowIterator.map(row => {
+            val bytes = projection.apply(row).getBytes
```

Comment:
why it needs to use UnsafeProjection to convert InternalRow to UnsafeRow? indeed, row is `UnsafeRow`.
Or 
```
            row match {
              case u: UnsafeRow => u.getBytes
              case i: InternalRow => projection.apply(i).getBytes
            }
```

Reply:
this would be better